SurfaceTopography/Uniform/common.py

```diff
@@ -225,10 +225,48 @@ def plot(topography, subplot_location=111):
     return ax
 
 
+def fill_undefined_data(topography, fill_value=-np.infty):
+    r"""
+    returns a topography where masked (undefined) data is replaced with
+    `fill_value`. For child topographies, note that this will also have the
+    same effect than `squeeze`
+
+    periodic, and info property will be transmitted to the returned topography
+
+    Parameters
+    ----------
+    topography: Topography or UniformLineScan instance
+    fill_value: float or array of floats
+        masked value in topography will be replaced
+    """
+    heights = topography.heights()
+    heights = np.ma.filled(heights, fill_value=fill_value)
+
+    if topography.dim == 1:
+        t = UniformLineScan(heights,
```

Comment:
Is this what we want? This is breaking the pipeline concept, as it is returning a new topography.

Reply:
I agree, but since the application is contact mechanics, we would make a `squeeze` afterwards anyway

Reply:
Still that should be at the discretion of the user. Please implement a proper pipeline class that handles this.

Reply:
Ok